internal/pkg/opentelemetry/controller.go

```diff
@@ -30,21 +30,35 @@ type Controller struct {
 	logger         logr.Logger
 	version        string
 	tracerProvider trace.TracerProvider
-	tracersMap     map[string]trace.Tracer
+	tracersMap     map[tracerID]trace.Tracer
 	bootTime       int64
 }
 
-func (c *Controller) getTracer(pkg string) trace.Tracer {
-	t, exists := c.tracersMap[pkg]
+type tracerID struct{ name, version, schema string }
+
+func (c *Controller) getTracer(pkg, tracerName, version, schema string) trace.Tracer {
+	tID := tracerID{name: pkg, version: c.version}
+	if tracerName != "" {
+		tID = tracerID{name: tracerName, version: version, schema: schema}
```

Comment:
nit

```suggestion
		tID.name = tracerName
```

Reply:
I don't think this is correct. The `version`  argument is the user-defined version we collected in the probe, and `c.version` is our distro version. So the assumption here is that if we have a user-provided tracer name we should also take the user-provided version and schema.
It is a bit confusing.